Added custom status in user profile and post header and review fixes

manojmalik20 opened the pull request:
Added custom status in user profile page
Added custom status emoji in post header
Added functionality to preventDoubleTap everywhere
Added localization ids
Review fixes

<!-- Thank you for contributing a pull request! Here are a few tips to help you:

1. If this is your first contribution, make sure you've read the Contribution Checklist https://developers.mattermost.com/contribute/getting-started/contribution-checklist/
2. Read our blog post about "Submitting Great PRs" https://developers.mattermost.com/blog/2019-01-24-submitting-great-prs
3. Take a look at other repository specific documentation at https://developers.mattermost.com/contribute
-->

#### Summary
<!--
A brief description of what this pull request does.
-->

#### Ticket Link
<!--
If this pull request addresses a Help Wanted ticket, please link the relevant GitHub issue, e.g.

  Fixes https://github.com/mattermost/mattermost-server/issues/XXXXX

Otherwise, link the JIRA ticket.
-->

#### Checklist
<!--
Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.
-->
- [ ] Added or updated unit tests (required for all new features)
- [ ] Has UI changes
- [ ] Includes text changes and localization file updates

#### Device Information
This PR was tested on: <!-- Device name(s), OS version(s) --> 

#### Screenshots
<!--
If the PR includes UI changes, include screenshots/GIFs (for both iOS and Android if possible).
-->

manojmalik20 committed:
Added custom status in user profile and post header and review fixes
Added custom status in user profile page
Added custom status emoji in post header
Added functionality to preventDoubleTap everywhere
Added localization ids
Review fixes

chetanyakan commented on app/screens/custom_status/custom_status_modal.tsx:
```diff
@@ -126,65 +127,33 @@ class CustomStatusModal extends PureComponent<Props, State> {
         const {recentCustomStatuses, theme} = this.props;
         const style = getStyleSheet(theme);
-        const recentStatuses = recentCustomStatuses.map((status: UserCustomStatus, index: number) => (
+        let arr = recentCustomStatuses;
```
Use meaningful name vs `arr`

chetanyakan commented on app/screens/custom_status/custom_status_modal.tsx:
```diff
@@ -289,8 +258,8 @@ class CustomStatusModal extends PureComponent<Props, State> {
                         {customStatusInput}
-                        {this.renderRecentCustomStatuses()}
-                        {this.renderCustomStatusSuggestions()}
+                        {this.renderSuggestions('recent')}
+                        {this.renderSuggestions('suggestions')}
```
1. Use constants here.
2. Check if using a map would work here:
```suggestion
                        {['recent', 'suggestions'].map(this.renderSuggestions)}
```

chetanyakan commented on app/screens/user_profile/user_profile.js:
```diff
@@ -372,6 +402,14 @@ const createStyleSheet = makeStyleSheetFromTheme((theme) => {
             flex: 1,
         },
+        iconContainer: {
+
+            // width: 45,
+            // height: 50,
```
Remove commented code.

manojmalik20 committed:
Fixed several issues
Increased touch area of clear button everywhere
Fixed the event propagation issue in clearing suggestions
Fixed the center alignment issue of title in custom status modal
Added clear button in user profile page
Review fixes

chetanyakan committed:
Fix types for custom status modal

manojmalik20 committed:
Merge branch 'MI-1190' of github.com:brightscout-alpha/mattermost-mobile into MI-1190

chetanyakan committed:
Fix tsc

manojmalik20 committed:
Fixed types and separated functions for Recent and suggestions in custom status modal

chetanyakan committed:
Merge branch 'MI-1190' of github.com:brightscout-alpha/mattermost-mobile into MI-1190

* 'MI-1190' of github.com:brightscout-alpha/mattermost-mobile:
  Fixed types and separated functions for Recent and suggestions in custom status modal

chetanyakan committed:
Update app/components/sidebars/settings/settings_sidebar_base.js

manojmalik20 committed:
Updated localization id in settings sidebar

manojmalik20 committed:
Changed the method in removeRecentCustomStatus API call
Changed names of some variables

chetanyakan commented on app/mm-redux/client/client4.ts:
```diff
@@ -846,7 +846,7 @@ export default class Client4 {
         return this.doFetch(
             `${this.getUserRoute('me')}/status/custom/recent`,
-            {method: 'delete', body: JSON.stringify(customStatus)},
+            {method: 'PATCH', body: JSON.stringify(customStatus)},
```
nit - all other methods are in lower case

manojmalik20 committed:
Merge branch 'master' of github.com:brightscout-alpha/mattermost-mobile into MI-1190

manojmalik20 committed:
Fixed the input height issue in ios and modalContext issue in ios

manojmalik20 committed:
Fixed various issues
Fixed clear button border radius issue in ios
Changed the method to post in removeRecentCustomStatus API call
Made the custom status modal input single line and fixed the height
Changed the height of iconContainer in custom status suggestion

chetanyakan committed:
Merge branch 'custom-status' into MI-1190
